exporter/otlpexporter/otlp.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpexporter
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	otlpmetrics "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/metrics/v1"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/trace/v1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+type otlpExporter struct {
+	exporters chan *exporterImp
+}
+
+type exporterErrorCode int
+type exporterError struct {
+	code exporterErrorCode
+	msg  string
+}
+
+var _ error = (*exporterError)(nil)
+
+func (e *exporterError) Error() string {
+	return e.msg
+}
+
+const (
+	defaultNumWorkers int = 8
+
+	_ exporterErrorCode = iota // skip 0
+	// errEndpointRequired indicates that this exporter was not provided with an endpoint in its config.
+	errEndpointRequired
+	// errAlreadyStopped indicates that the exporter was already stopped.
+	errAlreadyStopped
+)
+
+// NewTraceExporter creates an OTLP trace exporter.
+func NewTraceExporter(logger *zap.Logger, config configmodels.Exporter) (exporter.TraceExporter, error) {
+	oce, err := createOTLPExporter(logger, config)
+	if err != nil {
+		return nil, err
+	}
+	oexp, err := exporterhelper.NewTraceExporter(
+		config,
+		oce.pushTraceData,
+		exporterhelper.WithShutdown(oce.Shutdown))
+	if err != nil {
+		return nil, err
+	}
+
+	return oexp, nil
+}
+
+// NewMetricsExporter creates an OTLP metrics exporter.
+func NewMetricsExporter(logger *zap.Logger, config configmodels.Exporter) (exporter.MetricsExporter, error) {
+	oce, err := createOTLPExporter(logger, config)
+	if err != nil {
+		return nil, err
+	}
+	_, err = exporterhelper.NewMetricsExporter(
+		config,
+		oce.pushMetricsData,
+		exporterhelper.WithShutdown(oce.Shutdown),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: return created exporter after metric support is finalized in PushMetricsData()
+	return nil, configerror.ErrDataTypeIsNotSupported
+}
```

Comment:
Why do we not create v2 exporter/receiver for this?

Reply:
We do not have translation functions from OTLP to internal yet. We do have OTLP to OC translation function. To avoid delays and make OTLP available for Beta release this is the fastest way.